dev: clippy fails under --all-features and under --tests (pre-existing, outside CI gate)

## Summary

Two pre-existing clippy failures that are NOT caught by CI (CI runs `cargo clippy --locked -- -D warnings`, i.e. default features and no test code), but bite anyone running a broader clippy locally:

### 1. `--all-features`: OpenTelemetry `SpanBuilder` API mismatch
`cargo clippy -p freenet --all-features` fails to compile:

```
error[E0560]: struct `opentelemetry::trace::SpanBuilder` has no field named `span_id`
  --> crates/core/src/tracing.rs:2259
error[E0560]: struct `opentelemetry::trace::SpanBuilder` has no field named `trace_id`
  --> crates/core/src/tracing.rs:2260
```

The `trace-ot` feature pulls in an opentelemetry version whose `SpanBuilder` no longer exposes `span_id` / `trace_id` fields (available fields: `span_kind`, `events`, `links`). Either the dep needs pinning or the construction in `tracing.rs` needs updating to the current API.

### 2. `--tests`: wildcard_enum_match_arm in operations.rs
`cargo clippy --locked -p freenet --tests -- -D warnings` fails:

```
error: wildcard match will also match any future added variants
  --> crates/core/src/operations.rs:2332
  = note: `-D clippy::wildcard-enum-match-arm` implied by `-D warnings`
```

A `_ => ErrorKind::OperationError { .. }` arm over `OpError` trips the lint when test code is compiled. Either enumerate the variants or add a scoped `#[allow(clippy::wildcard_enum_match_arm)]` with a justifying comment.

## Why file rather than fix inline

Both are pre-existing on `main` and unrelated to the change that surfaced them (PR #4323, a config guard fix). Filing so they're tracked; whoever touches these areas (or wants to tighten the CI clippy invocation to include `--tests`) can clear them.

[AI-assisted - Claude]

--- a/crates/core/src/client_events.rs
+++ b/crates/core/src/client_events.rs
@@ -2333,10 +2333,17 @@ pub(crate) mod test {
         // this test catches the mismatch via the variant-equality
         // assertion below.
         let err = OpError::NodeShuttingDown;
+        // The wildcard arm below intentionally absorbs every current (and any
+        // future) `OpError` variant the way production's `report_op_init_error`
+        // catch-all does; this test only pins the `NodeShuttingDown` arm, so an
+        // exhaustive listing would add churn without strengthening the
+        // assertion. Mirrors the documented `non_exhaustive` wildcard pattern in
+        // `.claude/rules/git-workflow.md`. The allow sits on the match
+        // expression because clippy reports `wildcard_enum_match_arm` at the
+        // match scrutinee, not the individual arm.
+        #[allow(clippy::wildcard_enum_match_arm)]
         let kind = match err {
             OpError::NodeShuttingDown => ErrorKind::Shutdown,
-            // All other variants fall into the catch-all in
-            // production — they're irrelevant to this test.
             _ => ErrorKind::OperationError {
                 cause: "irrelevant for this test".to_string().into(),
             },
@@ -2370,15 +2377,24 @@ pub(crate) mod test {
         let instance_id = ContractInstanceId::new([7u8; 32]);
         let err = OpError::ContractBanned { instance_id };
         let op_name = "PUT";
+        // The wildcard arms below intentionally absorb every other (and any
+        // future) `OpError` / `ErrorKind` variant the way production's
+        // catch-alls do; this test only pins the `ContractBanned` mapping, so an
+        // exhaustive listing would add churn without strengthening the
+        // assertion. Mirrors the documented non_exhaustive wildcard pattern in
+        // `.claude/rules/git-workflow.md`. The allows sit on the match
+        // expressions because clippy reports `wildcard_enum_match_arm` at the
+        // match scrutinee, not the individual arm.
+        #[allow(clippy::wildcard_enum_match_arm)]
         let kind = match err {
             OpError::ContractBanned { .. } => ErrorKind::OperationError {
                 cause: format!("{op_name} operation failed: {err}").into(),
             },
-            // Other variants are irrelevant to this test.
             _ => ErrorKind::Unhandled {
                 cause: "irrelevant for this test".to_string().into(),
             },
         };
+        #[allow(clippy::wildcard_enum_match_arm)]
         match kind {
             ErrorKind::OperationError { cause } => {
                 assert!(

--- a/crates/core/src/server/app_packaging.rs
+++ b/crates/core/src/server/app_packaging.rs
@@ -195,6 +195,51 @@ impl WebApp {
     }
 }
 
+impl<'a> TryFrom<&'a [u8]> for WebApp {
+    type Error = WebContractError;
+
+    fn try_from(state: &'a [u8]) -> Result<Self, Self::Error> {
+        debug!(
+            "Attempting to create WebApp from {} bytes of state",
+            state.len()
+        );
+        const MAX_METADATA_SIZE: u64 = 1024;
+        const MAX_WEB_SIZE: u64 = 1024 * 1024 * 100;
+        // Decompose the state and extract the compressed web interface
+        let mut state = Cursor::new(state);
+
+        let metadata_size = state
+            .read_u64::<BigEndian>()
+            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
+        if metadata_size > MAX_METADATA_SIZE {
+            return Err(WebContractError::UnpackingError(anyhow::anyhow!(
+                "Exceeded metadata size of 1kB: {} bytes",
+                metadata_size
+            )));
+        }
+        let mut metadata = vec![0; metadata_size as usize];
+        state
+            .read_exact(&mut metadata)
+            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
+
+        let web_size = state
+            .read_u64::<BigEndian>()
+            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
+        if web_size > MAX_WEB_SIZE {
+            return Err(WebContractError::UnpackingError(anyhow::anyhow!(
+                "Exceeded packed web size of 100MB: {} bytes",
+                web_size
+            )));
+        }
+        let mut web = vec![0; web_size as usize];
+        state
+            .read_exact(&mut web)
+            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
+
+        Ok(Self { metadata, web })
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
@@ -341,50 +386,5 @@ mod tests {
             std::fs::read(dst.path().join("assets/app.js")).unwrap(),
             b"console.log(1)"
         );
-    }
-}
-
-impl<'a> TryFrom<&'a [u8]> for WebApp {
-    type Error = WebContractError;
-
-    fn try_from(state: &'a [u8]) -> Result<Self, Self::Error> {
-        debug!(
-            "Attempting to create WebApp from {} bytes of state",
-            state.len()
-        );
-        const MAX_METADATA_SIZE: u64 = 1024;
-        const MAX_WEB_SIZE: u64 = 1024 * 1024 * 100;
-        // Decompose the state and extract the compressed web interface
-        let mut state = Cursor::new(state);
-
-        let metadata_size = state
-            .read_u64::<BigEndian>()
-            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
-        if metadata_size > MAX_METADATA_SIZE {
-            return Err(WebContractError::UnpackingError(anyhow::anyhow!(
-                "Exceeded metadata size of 1kB: {} bytes",
-                metadata_size
-            )));
-        }
-        let mut metadata = vec![0; metadata_size as usize];
-        state
-            .read_exact(&mut metadata)
-            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
-
-        let web_size = state
-            .read_u64::<BigEndian>()
-            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
-        if web_size > MAX_WEB_SIZE {
-            return Err(WebContractError::UnpackingError(anyhow::anyhow!(
-                "Exceeded packed web size of 100MB: {} bytes",
-                web_size
-            )));
-        }
-        let mut web = vec![0; web_size as usize];
-        state
-            .read_exact(&mut web)
-            .map_err(|e| WebContractError::UnpackingError(anyhow::anyhow!(e)))?;
-
-        Ok(Self { metadata, web })
     }
 }

--- a/crates/core/src/tracing.rs
+++ b/crates/core/src/tracing.rs
@@ -2233,8 +2233,8 @@ mod opentelemetry_tracer {
 
     use dashmap::DashMap;
     use opentelemetry::{
-        KeyValue, global,
-        trace::{self, Span},
+        Context, KeyValue, global,
+        trace::{self, Span, TraceContextExt},
     };
 
     use super::*;
@@ -2253,17 +2253,31 @@ mod opentelemetry_tracer {
             let mut span_id = [0; 8];
             span_id.copy_from_slice(&tx_bytes[8..]);
             let start_time = transaction.started();
-            let inner = tracer.build(trace::SpanBuilder {
-                name: transaction.transaction_type().description().into(),
-                start_time: Some(start_time),
-                span_id: Some(trace::SpanId::from_bytes(span_id)),
-                trace_id: Some(trace::TraceId::from_bytes(tx_bytes)),
-                attributes: Some(vec![
-                    KeyValue::new("transaction", transaction.to_string()),
-                    KeyValue::new("tx_type", transaction.transaction_type().description()),
-                ]),
-                ..Default::default()
-            });
+            // opentelemetry 0.32 removed the `trace_id`/`span_id` fields from
+            // `SpanBuilder`; trace identity is now seeded from the parent
+            // `Context`. We anchor the span on a deterministic remote
+            // `SpanContext` derived from the transaction bytes so all events of
+            // a transaction continue to share a stable trace_id. The child span
+            // receives a fresh span_id from the SDK id generator (no longer
+            // settable through the public API), with our deterministic span_id
+            // recorded as the parent span_id.
+            let parent_span_context = trace::SpanContext::new(
+                trace::TraceId::from_bytes(tx_bytes),
+                trace::SpanId::from_bytes(span_id),
+                trace::TraceFlags::SAMPLED,
+                true,
+                trace::TraceState::default(),
+            );
+            let parent_cx = Context::current().with_remote_span_context(parent_span_context);
+            let builder = trace::SpanBuilder::from_name(
+                transaction.transaction_type().description().to_string(),
+            )
+            .with_start_time(start_time)
+            .with_attributes(vec![
+                KeyValue::new("transaction", transaction.to_string()),
+                KeyValue::new("tx_type", transaction.transaction_type().description()),
+            ]);
+            let inner = tracer.build_with_context(builder, &parent_cx);
             OTSpan {
                 inner,
                 last_log: SystemTime::now(),

--- a/crates/core/src/transport/peer_connection.rs
+++ b/crates/core/src/transport/peer_connection.rs
@@ -3474,7 +3474,7 @@ mod tests {
 
         let samples_after = crate::transport::TRANSPORT_METRICS.rtt_sample_count();
         assert!(
-            samples_after >= samples_before + 1,
+            samples_after > samples_before,
             "Pong for a pending ping must record at least one RTT sample \
              (before={samples_before}, after={samples_after}). \
              The keep-alive RTT path (#4000) is not wired up."

--- a/crates/core/src/transport/sent_packet_tracker.rs
+++ b/crates/core/src/transport/sent_packet_tracker.rs
@@ -567,6 +567,16 @@ impl SentPacketTracker<RealTime> {
 // Unit tests
 #[cfg(test)]
 pub(in crate::transport) mod tests {
+    // These tests assert that `get_resend()` returns a specific `ResendAction`
+    // variant and `panic!` on anything else (`match action { Expected => ..,
+    // _ => panic!() }`). That wildcard-with-panic is the intended assertion
+    // idiom — expanding it into a `pat | _` listing would only add churn and
+    // would still need updating for any new variant. Scope the allow to the
+    // test module rather than annotating each identical arm. Mirrors the
+    // documented non_exhaustive wildcard pattern in
+    // `.claude/rules/git-workflow.md`.
+    #![allow(clippy::wildcard_enum_match_arm)]
+
     use super::*;
     use crate::simulation::VirtualTime;
     use rstest::rstest;

